### PR TITLE
Updating governance to point to Oversight Committee members

### PR DESCRIPTION
**Pull request by joesepi**
Surfacing current list of oversight committee members as a community member wasn't able to easily find current membership.

### CODE REVIEW

**Review by russellb — APPROVED** [review]

File: GOVERNANCE.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ Advancement to the project Maintainer position, removal or stepping down, and du
 
 The initial Oversight Committee at the launch of the project was appointed by the founding sponsors of the InstructLab project. This bootstrap committee will serve until the first election of the Oversight Committee using processes and timing as determined by this group.
 
+The list of Oversight Committee members can be found [here](https://github.com/instructlab/community/blob/main/MAINTAINERS.md#oversight-committee).
```
Comment (russellb) [inline GOVERNANCE.md:36]: slight preference not to use `here` as a link target

```suggestion
The list of Oversight Committee members can be found in [MAINTAINERS.md](https://github.com/instructlab/community/blob/main/MAINTAINERS.md#oversight-committee).
```

Comment (lhawthorn) [reply GOVERNANCE.md:36]: Agree with Russell's suggestion.

**Review by lhawthorn — COMMENTED** [review]
With the feedback from @russellb addressed, I believe this change is good to merge. Thank you for the contribution, @joesepi

File: GOVERNANCE.md (hunk 1 above)
Comment (joesepi) [reply GOVERNANCE.md:36]: Yeah, I agree. I was following the pattern in the previous section. Will accept suggestion.

**Review by russellb — APPROVED** [review]

**Pull request merged**